src/qproc/QuerySession.h

```diff
@@ -142,8 +142,12 @@ class QuerySession {
 
     std::shared_ptr<query::SelectStmt> getMergeStmt() const;
 
+    /// Build the template for the worker queries.
+    /// @param fillinChunIdTag - When true replace the CHUNK_TAG string with the chunk id number.
+    ///         Template strings sent to the worker should not fill in the tag, but unit tests
+    ///         need it filled in.
     ChunkQuerySpec::Ptr buildChunkQuerySpec(query::QueryTemplate::Vect const& queryTemplates,
-                                            ChunkSpec const& chunkSpec) const;
+                                            ChunkSpec const& chunkSpec, bool fillInChunkIdTag = true) const;
 
```

Comment:
I did it this way as there is only one location in the code where this function is called to produce a template string where the tag is not replaced with the chunk id, but there are several unit tests that do. Chainging that is simple. The existing unit test would fail if the template construction or subtitution had an error, so I didn't think new test were needed.

Reply:
Thanks! In general, it's better (in the long run) to reduce or avoid modifying the normal flow of the code for improving the unit tests. I'd rather modify the unit tests to make them happy.